nomad/job_endpoint_validators_test.go

```diff
@@ -0,0 +1,83 @@
+package nomad
+
+import (
+	"testing"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestJobNamespaceConstraintCheckHook_Name(t *testing.T) {
+	t.Parallel()
+
+	require.Equal(t, "namespace-constraint-check", new(jobNamespaceConstraintCheckHook).Name())
+}
+
+func TestJobNamespaceConstraintCheckHook_taskValidateDriver(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		description string
+		driver      string
+		ns          *structs.Namespace
+		result      bool
+	}{
+		{
+			"No drivers enabled/disabled, allow all",
+			"docker",
+			&structs.Namespace{},
+			true,
+		},
+		{
+			"Only exec and docker are allowed 1/2",
+			"docker",
+			&structs.Namespace{
+				Capabilities: &structs.NamespaceCapabilities{
+					EnabledTaskDrivers: []string{"docker", "exec"}},
+			},
+			true,
+		},
+		{
+			"Only exec and docker are allowed 2/2",
+			"raw_exec",
+			&structs.Namespace{
+				Capabilities: &structs.NamespaceCapabilities{
+					EnabledTaskDrivers: []string{"docker", "exec"}},
+			},
+			false,
+		},
+		{
+			"disable takes precedence over enable", // TODO: Wanted?
```

Comment:
Yeah, let's keep this test. It might be nice to validate that we don't have collisions when writing the namespace in `(*Namespace) Validate`, but it's probably a good idea to enforce that `disable` takes precedence on job submission too.